### ISSUE TITLE
How many hours of data did you use for training and testing? What is the average duration of each audio file?

### ISSUE THREAD

**Issue by syedasaraCodeNinja** [Bushramjad/Speech-Recognition-in-Urdu#3]

**Comment by Bushramjad** [comment]
It's about 7 hours of recorded data, which was then split for training and testing.

**Comment by syedasaraCodeNinja** [comment]
Average duration of each audio?

**Comment by syedasaraCodeNinja** [comment]
Did your model give reasonable predictions because the training loss is NAN in both steps?

**Comment by Bushramjad** [comment]
@syedasaraCodeNinja 2 sec each

**Comment by Asifm93** [comment]
I am training similar model to this model on other local language using colab pro 40 GB GPU still it says out of memory error ; Training data is around 4-7 hours. Can you connect with me to help out to your extent ; asifrehman982@gmail.com

**Comment by syedasara98** [comment]
What is the duration of your each audio file?

> I am training similar model to this model on other local language using colab pro 40 GB GPU still it says out of memory error ; Training data is around 4-7 hours. Can you connect with me to help out to your extent ; [asifrehman982@gmail.com](mailto:asifrehman982@gmail.com)

**Comment by Asifm93** [comment]
Min 2 sec max 55 sec

**Comment by syedasara98** [comment]
55 seconds is too large. The duration of audio file should be from 2 secs to 8 secs. 
Secondly, change the "per_device_train_batch_size" from 16 to 8 or 4. The memory error could be because of large batch size

**Comment by Asifm93** [comment]
Yes tried different options still getting same can you have look at code if possible ?

**Comment by syedasara98** [comment]
Yeah sure.

**Comment by Asifm93** [comment]
alright just email when you have enough time thank you  ;  asifrehman982@gmail.com

**Comment by Asifm93** [comment]
> Yeah sure.

??

**Comment by Asifm93** [comment]
A.o.a,
Hope You'll be doing well. Kindly do a favor to look at code and guide
what's the issue if possible.


Thank you :)
Kind regards.
